src/EventStoreTools/EventStoreConnectionBuilder.cs

```diff
@@ -0,0 +1,69 @@
+﻿// <copyright file="EventStoreConnectionBuilder.cs" company="Cognisant">
+// Copyright (c) Cognisant. All rights reserved.
+// </copyright>
+
+namespace Cr.EventStoreTools
+{
+    using System;
+    using EventStore.ClientAPI;
+
+    /// <summary>
+    /// A helper class for quickly building an <see cref="IEventStoreConnection"/>.
+    /// </summary>
+    // ReSharper disable once UnusedMember.Global
+    public static class EventStoreConnectionBuilder
+    {
+        /// <summary>
+        /// Build an <see cref="IEventStoreConnection"/> using the passed in connection <see cref="Uri"/> and configuration.
+        /// </summary>
+        /// <param name="connection">The URI for the Event Store's external TCP port, including login details.</param>
+        /// <param name="retryLimit">An optional maximum allowed retries for any given operation; this defaults to 10.</param>
+        /// <param name="gossipTimeout">An optional gossip timeout for the EventStore connection; this defaults to 5 seconds.</param>
+        /// <param name="logger">An optional <see cref="ILogger"/> to use to log information about and issues with the <see cref="IEventStoreConnection"/>.</param>
+        /// <returns>A configured, connected <see cref="IEventStoreConnection"/> which uses any passed in configuration variables, or the defaults detailed in the parameters section.</returns>
+        // ReSharper disable once UnusedMember.Global
+        public static IEventStoreConnection SetUpEventStoreConnection(Uri connection, int retryLimit = 10, TimeSpan? gossipTimeout = null, ILogger logger = null)
+        {
```

Comment:
We stated on the Trello card you should optionally be able to pass in the Event (`EventStoreClientOnDisconnected`, `EventStoreClientOnConnected`, etc.)